### ISSUE TITLE
Fedora 32: screen lock not working any more when gtile is active

### ISSUE THREAD

**Issue by dhiller** [gTile/gTile#132]
Since todays latest system update the screen lock (either using SUPER+L or clicking "Lock" from system menu stopped working.

Symptom: Whenever I try to use it, the Problem Reporting reports a `gnome-shell killed by SIGABRT`

kernel: 5.6.8-300.fc32.x86_64

Stacktrace from Problem details `/var/log/messages`:

```
May 04 13:32:31 dhiller-fedora-work gnome-shell[9763]: ../clutter/clutter/clutter-actor.c:10548: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
May 04 13:32:33 dhiller-fedora-work gnome-shell[9763]: g_signal_handler_disconnect: assertion 'handler_id > 0' failed
May 04 13:32:48 dhiller-fedora-work gnome-shell[9763]: g_signal_handler_disconnect: assertion 'handler_id > 0' failed
May 04 13:32:49 dhiller-fedora-work gnome-shell[9763]: loading default theme (Adwaita)
May 04 13:33:00 dhiller-fedora-work gnome-shell[9763]: loading default theme (Adwaita)
May 04 13:33:04 dhiller-fedora-work gnome-shell[9763]: [System monitor] applet enabling
May 04 13:33:04 dhiller-fedora-work gnome-shell[9763]: [System monitor] applet enabling done
May 04 13:33:06 dhiller-fedora-work gnome-shell[9763]: ../clutter/clutter/clutter-actor.c:10548: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
May 04 13:33:07 dhiller-fedora-work gnome-shell[9763]: Source ID 17572 was not found when attempting to remove it
May 04 13:33:07 dhiller-fedora-work gnome-shell[9763]: [System monitor] applet disable
May 04 13:33:16 dhiller-fedora-work gnome-shell[9763]: [System monitor] applet enabling
May 04 13:33:16 dhiller-fedora-work gnome-shell[9763]: [System monitor] applet enabling done
May 04 13:33:20 dhiller-fedora-work gnome-shell[9763]: Source ID 18462 was not found when attempting to remove it
May 04 13:33:20 dhiller-fedora-work gnome-shell[9763]: [System monitor] applet disable
May 04 13:33:53 dhiller-fedora-work gnome-shell[9763]: ../clutter/clutter/clutter-actor.c:10548: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Enabling begin
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Init gnome-shell version 3.36.2 major 3 minor 36
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Init settings
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile grid-sizes set to 4x6,6x6,9x9,2x2
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Starting...
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile initGrids
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile New Grid for monitor 0
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ToggleSettingsButton _update event animation
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ToggleSettingsButton Connect button-press-event
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ToggleSettingsButton Connect update-toggle
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ToggleSettingsButtonListener Connect update-toggle
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ToggleSettingsButton _update event auto-close
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ToggleSettingsButton Connect button-press-event
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ToggleSettingsButton Connect update-toggle
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ToggleSettingsButtonListener Connect update-toggle
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ActionButton Connect button-press-event
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile AutoTileMainAndList connect button-press-event
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile ActionButton Connect button-press-event
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile AutoTileTwoList connect button-press-event
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile hide true
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Connect hide-tiling for monitor 0
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Init grid done
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Create Button
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile GTileStatusButton _init done
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Binding keys
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Binding keys
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Binding keys
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: gTile Extention Enabled!
May 04 13:34:41 dhiller-fedora-work gnome-shell[9763]: Window manager warning: Overwriting existing binding of keysym ff0d with keysym ff0d (keycode 24).
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: **
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: St:ERROR:../src/st/st-theme.c:1048:_st_theme_resolve_url: assertion failed: (base_file)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: Bail out! St:ERROR:../src/st/st-theme.c:1048:_st_theme_resolve_url: assertion failed: (base_file)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: == Stack trace for context 0x55a75e60a920 ==
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #0   55a76129e710 i   resource:///org/gnome/shell/ui/extensionSystem.js:107 (2c258f868358 @ 467)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #1   55a76129e680 i   resource:///org/gnome/shell/ui/extensionSystem.js:593 (2c258f86a538 @ 15)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #2   7fff849c8710 b   self-hosted:266 (24ea8e5f3cb8 @ 259)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #3   55a76129e5f0 i   resource:///org/gnome/shell/ui/extensionSystem.js:592 (2c258f86a4c0 @ 98)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #4   55a76129e560 i   resource:///org/gnome/shell/ui/extensionSystem.js:610 (2c258f86a5b0 @ 82)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #5   7fff849c99a0 b   self-hosted:1009 (19096148e088 @ 423)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #6   7fff849c9aa0 b   resource:///org/gnome/gjs/modules/core/_signals.js:133 (2f853b71f538 @ 427)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #7   7fff849ca5c0 b   resource:///org/gnome/shell/ui/sessionMode.js:195 (2f853b724d30 @ 286)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #8   55a76129e400 i   resource:///org/gnome/shell/ui/sessionMode.js:156 (2f853b724b50 @ 40)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #9   55a76129e348 i   resource:///org/gnome/shell/ui/screenShield.js:571 (2f853b70fd30 @ 191)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #10   55a76129e288 i   resource:///org/gnome/shell/ui/screenShield.js:620 (2f853b70fda8 @ 419)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #11   55a76129e1e8 i   resource:///org/gnome/shell/misc/systemActions.js:417 (1909614559e8 @ 88)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #12   55a76129e158 i   resource:///org/gnome/shell/ui/status/system.js:106 (2c255d12dcb8 @ 53)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #13   55a76129e0b8 i   resource:///org/gnome/shell/ui/popupMenu.js:191 (2c258f81b178 @ 25)
May 04 13:34:43 dhiller-fedora-work gnome-shell[9763]: #14   55a76129e018 i   resource:///org/gnome/shell/ui/popupMenu.js:138 (2c258f814e98 @ 88)
May 04 13:34:44 dhiller-fedora-work gnome-shell[11710]: Unknown experimental feature 'x11-randr-fractional-scaling'
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: Unset XDG_SESSION_ID, getCurrentSessionProxy() called outside a user session. Asking logind directly.
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: Will monitor session 2
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: Telepathy is not available, chat integration will be disabled.
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile Enabling begin
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile Init gnome-shell version 3.36.2 major 3 minor 36
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile Init settings
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile grid-sizes set to 4x6,6x6,9x9,2x2
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile Starting...
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile initGrids
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile New Grid for monitor 0
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton _update event animation
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton Connect button-press-event
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton Connect update-toggle
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButtonListener Connect update-toggle
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton _update event auto-close
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton Connect button-press-event
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton Connect update-toggle
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButtonListener Connect update-toggle
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ActionButton Connect button-press-event
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile AutoTileMainAndList connect button-press-event
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ActionButton Connect button-press-event
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile AutoTileTwoList connect button-press-event
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile hide true
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile Connect hide-tiling for monitor 0
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile New Grid for monitor 1
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton _update event animation
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton Connect button-press-event
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton Connect update-toggle
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButtonListener Connect update-toggle
May 04 13:34:46 dhiller-fedora-work gnome-shell[11710]: gTile ToggleSettingsButton _update event auto-close
```

Disabling the extension solves the issue.

**Comment by dhiller** [comment]
Screenshot of Problem Reporting: https://snipboard.io/blEnX8.jpg

**Comment by scherepanov** [comment]
Works just fine with me, on same environment.
Very strange.

**Comment by floriankisser** [comment]
Also got SIGABRT on screen lock since today. Fedora 32, currently Xorg.

**Comment by scherepanov** [comment]
Well... I have identical environment, freshly updated Fedora 32.
I can lock/unlock, no problems...
Though, reviewing messages, I found one gnome-shell crash immediately after recent update last night.
After power cycle it never repeats again.
Cannot reproduce :-(

**Comment by bzikarsky** [comment]
I have the same issue. All extensions disabled besides gTile. :grimacing: 

Hm. Interesting. I reloaded the extension once (`gnome-extensions disable gTile@vibou && gnome-extensions enable gTile@vibou`) and the issue is gone. (At least for now. I'll check later with log-off, log-on).

Maybe related: Sometimes gTile does seem to detect the wrong screen geometry and does not show its dialog centered. This can also be fixed with an extension reload and only happens after a fresh login. I always assumed this is because of the screen configuration for my user (1440x2560 (rotated) - 3840x1600 (primary) - 2560x1440) which only gets applied after login.

**Comment by bzikarsky** [comment]
Update: The error was back after a fresh login. Took a number of extension reloads to be fixed again.

I then created an X11 conf which reflects my current xrandr config. After reboot it seems to work without crashing now. So I guess, it's somehow related to the screen config changed after login? :shrug:

**Comment by scherepanov** [comment]
That is correct, screen config is caclulated on gTile enable.
If you run randr config, change in screen config is not reflected in gTile.
This should be fixable by attaching to on_monitor_change signal. I tried to find it in past, but cannot figure out correct signal.
If you can find what signal is being emitted in gnome-shell, I can issue a fix for screen config change.
For crashes, I still do not have any problems. Reviewing history of crashes on my PC since upgrade to Fedora 32, I do see one gnome-shell crash, but after reboot it never show up again. Crash was on St library styling, I would expect it fixed in recent gnome-shell 3.36.2.
Basically, until I can reproduce this crash, it is not going to be fixed. Unless someone else will look on it too.

**Comment by floriankisser** [comment]
FYI here is the retrace report
https://retrace.fedoraproject.org/faf/reports/2904053/

**Comment by scherepanov** [comment]
Looks like St library is having a lot of crashes. Hopefully St maintainers will fix it, and it will fix gTile-induced gnome-shell crashes.
I would try to play with styling and see what is causing crash.... but I do not have crashes on my box.

**Comment by sirianni** [comment]
Here's a [bug report in gnome-shell](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2757) for this same issue

**Comment by scherepanov** [comment]
Yes, I was able to reproduce.

libst is throwing sigabrt on failed assert on _st_theme_resolve_url
Seems to be cannot find some icon image.

May 09 16:55:32 supermedia gnome-shell[6367]: St:ERROR:../src/st/st-theme.c:1048:_st_theme_resolve_url: assertion failed: (base_file)
May 09 16:55:32 supermedia gnome-shell[6367]: Bail out! St:ERROR:../src/st/st-theme.c:1048:_st_theme_resolve_url: assertion failed: (base_file)

And, it happens sporadically.
That would not be an easy fix.
gTile has been written a while ago, it is using more advanced manual control of style...

**Comment by scherepanov** [comment]
It comes down to bug in libst-1.0
Here is a bug I reported:
https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2768
It is possible to do workaround, by removing nice icon animation. I prefer gnome-shell team fixing bug in libst

**Comment by scherepanov** [comment]
Good news - fix issued in upstream in gnome-shell repo on gitlab.
Now need to wait when it will be merged into master, and when it will be released into wild.

**Comment by fastcat** [comment]
FWIW, this affects Debian Bullseye too. There though it seems to segfault instead of aborting, and also locking after the crashed gnome-shell restarts does work.

As noted in one of the upstream reports for gnome-shell, it seems to be related at least partly to how many times you lock/unlock the screen since gnome-shell (re)started. First time seems to be OK at least, after that, depends.

**Comment by scherepanov** [comment]
Yes it affects all disto with gnome-shell 3.36.2
Fix already in work upstream. You can wait till it will be released and pushed to your disto, Or you can get patch and build gnome-shell with a fix.

**Comment by fastcat** [comment]
A note from upstream regarding the fix (https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1247#note_806252):

> it's almost certainly an extension bug if we hit that code path

Not sure if there's something special about gTile here, or a use case upstream hasn't considered? Or I may just be misinterpreting the context of the comment

**Comment by scherepanov** [comment]
gnome-shell developers are discussing how to combine stylesheets from elements where one element is a child of another, and both has stylesheets.
This is exactly how gTile has been written many years ago.
Used to be working just fine, until recent libst-1.0 big rewrite.
I do not see any reason why it should not be working.
Discussion there is about edge case with conflicting stylesheet settings. It appears gTile is not defining stylesheets that way, looks completely normal to me.
Anyway, it appears it will be a log entry when this code path will be executed. Will find when gnome-shell3.36.3 will be out. Should be "soon".

**Comment by scherepanov** [comment]
Just updated Fedora32, and gnome-shell works properly now. Screen locks are not causing crashes.
Looks like bug is fixed in gnome-shell 3.36.2-3
Can someone verify?
